crates/deltalake-core/src/table/state.rs

```diff
@@ -41,7 +41,8 @@ pub struct DeltaTableState {
     app_transaction_version: HashMap<String, i64>,
     // table metadata corresponding to current version
     current_metadata: Option<DeltaTableMetaData>,
```

Comment:
Should this still be there? Should the old fn metadata point to the new metadata field instead of this one? So you don't store it twice